include/rocksdb/utilities/write_batch_with_index.h

```diff
@@ -359,6 +377,12 @@ class WriteBatchWithIndex : public WriteBatchBase {
                                  ColumnFamilyHandle* column_family,
                                  const Slice& key, PinnableWideColumns* columns,
                                  ReadCallback* callback);
+  void MultiGetEntityFromBatchAndDB(DB* db, const ReadOptions& read_options,
```

Comment:
I have a question for my own learning.

I'm wondering why we have these APIs with `ReadCallback*` (not just `MultiGetEntityFromBatchAndDB`, but also `GetEntityFromBatchAndDB`, `MultiGetFromBatchAndDB` and `GetFromBatchAndDB`)

I see that we simply call them from the public facing one with `nullptr` callback. These are under `private:`, so I wasn't sure if these are meant to be public APIs or I'm missing something here.

Reply:
The `ReadCallback`s of these methods are utilized by the transaction layer; the corresponding classes are marked as `friend`s of `WriteBatchWithIndex` so they can access these private methods.

Reply:
Aha! Thank you. I guess that's why these ones have assertion only instead of checking the input + returning `Status::InvalidArgument`